sql_assesment.txt

```diff
@@ -0,0 +1,1786 @@
+
+
+select count( distinct city) from customers where  city is not null;
+
++-----------------------+
+| count( distinct city) |
++-----------------------+
+|                     4 |
++-----------------------+
+
+
+
+83) Find the average amount from the Orders Table. 
+
+
+select avg(amt) from orders;
+
++----------+
+| avg(amt) |
++----------+
+|  2665.84 |
++----------+
+
+
+
+84) Find all customers who are not located in SanJose and whose rating is above 200. 
+
+
+
+select * from customers where city <> "Sanjose" and rating>200;
+
++------+-------+--------+--------+------+
+| CNUM | CNAME | CITY   | RATING | SNUM |
++------+-------+--------+--------+------+
+| 2004 | Grass | Berlin |    300 | 1002 |
++------+-------+--------+--------+------+
+
+
+
+
+85) Give a simpler way to write this query.SELECT snum, sname, city, comm FROM salespeople WHERE (comm > + 0.12 OR comm &lt; 0.14); 
+
+select * from salespeople where comm>12 or comm<14;
+
++------+---------+------------+------+
+| SNUM | SNAME   | CITY       | COMM |
++------+---------+------------+------+
+| 1001 | Peel    | London     |   12 |
+| 1002 | Serres  | SanJose    |   13 |
+| 1003 | AxelRod | New York   |   10 |
+| 1004 | Motika  | London     |   11 |
+| 1007 | Rofkin  | Barcenlona |   15 |
+| 1008 | Fran    | London     |   25 |
++------+---------+------------+------+
+
+
+
+86) Which salespersons attend to customers not in the city they have been assigned to? 
+
+
+select * from salespeople where snum in (select snum from customers where salespeople.city <> customers.city);
+
++------+---------+------------+------+
+| SNUM | SNAME   | CITY       | COMM |
++------+---------+------------+------+
+| 1002 | Serres  | SanJose    |   13 |
+| 1003 | AxelRod | New York   |   10 |
+| 1004 | Motika  | London     |   11 |
+| 1007 | Rofkin  | Barcenlona |   15 |
++------+---------+------------+------+
+
+
+
+
+87) Which salespeople get commission greater than 0.11 are serving customers rated less than 250? 
+
+
+select sname,comm,rating from salespeople, customers where salespeople.snum= customers.snum and rating<250  and comm>11;
+
++--------+------+--------+
+| sname  | comm | rating |
++--------+------+--------+
+| Peel   |   12 |    100 |
+| Peel   |   12 |    100 |
+| Serres |   13 |    200 |
++--------+------+--------+
+
+
+
+
+88) Which salespeople have been assigned to the same city but get different commission percentages? 
+
+select s1.snum,s1.sname,s2.sname from salespeople s1, salespeople s2 where s1.snum != s2.snum and  s1.city= s2.city and s1.snum<s2.snum group by snum,s1.sname,s2.sname;
+
++------+--------+--------+
+| snum | sname  | sname  |
++------+--------+--------+
+| 1001 | Peel   | Motika |
+| 1001 | Peel   | Fran   |
+| 1004 | Motika | Fran   |
++------+--------+--------+
+
+
+
+
+89) Which salesperson has earned the maximum commission? 
+
+
+select sname from salespeople where comm = (select max(comm) from salespeople);
```

Comment:
Here the maximum commission means how much a salesperson has earned on his all orders. Please correct it.